sessionctx/stmtctx/stmtctx.go

```diff
@@ -197,6 +197,9 @@ type StatementContext struct {
 	EnableOptimizeTrace bool
 	// LogicalOptimizeTrace indicates the trace for optimize
 	LogicalOptimizeTrace *tracing.LogicalOptimizeTracer
+	// IsNewlyInserted indicates the current statement is inserting new rows. This flag will be
+	// used to optimize the `delete-your-write` situation.
+	IsNewlyInserted bool
```

Comment:
Does this flag need to be reset when a new statement comes?

Reply:
It's needed, a new statement may create a new `stmtCtx` object, but for `StatementContext.ResetForRetry`, seems we still need to reset it there?

Reply:
Is the flag only for passing value from `AddRecord` to `index.Create`? I didn't under stand 😕 What if `insert on duplicate update` two records, one of which already exists?

Reply:
Yes, as the `index.Create` could be used by different upper functions such as `ReBuildIndex` we need to find out a way to classify if we are creating newly inserted index keys.

Reply:
> Is the flag only for passing value from `AddRecord` to `index.Create`? I didn't under stand 😕 What if `insert on duplicate update` two records, one of which already exists?

@MyonKeminta 

Seems there's still some problems and it's not statement level flag and should be row level, thanks for reminding.

Reply:
@you06 @MyonKeminta 
PTAL again, the `stmtCtx` is removed and create options is used so that the affect is not statement level.